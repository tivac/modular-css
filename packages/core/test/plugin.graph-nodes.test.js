"use strict";

var Graph   = require("dependency-graph").DepGraph,
    
    plugin   = require("../plugins/graph-nodes.js"),
    resolver = require("../lib/resolve.js").resolve,
    relative = require("../lib/relative.js"),
    
    processor = require("postcss")([ plugin ]);
    
function resolve(src, file) {
    return relative(process.cwd(), resolver(src, file));
}

describe("/plugins", function() {
    describe("/graph-nodes.js", function() {
        it("should populate a graph with external @value references", () => {
            var graph = new Graph(),
                from  = "packages/core/test/specimens/simple.css";

            graph.addNode(from);
            
            return processor.process(
                `@value one from "./local.css";`,
                { from, graph, resolve }
            )
            .then(() => expect(graph.overallOrder()).toMatchSnapshot());
        });

        it("should populate a graph with external composes references", () => {
            var graph = new Graph(),
                from  = "packages/core/test/specimens/simple.css";

            graph.addNode(from);
            
            return processor.process(
                `.a { composes: booga from "./local.css"; }`,
                { from, graph, resolve }
            )
            .then(() => expect(graph.overallOrder()).toMatchSnapshot());
        });

        it("should populate a graph with :external references", () => {
            var graph = new Graph(),
                from  = "packages/core/test/specimens/simple.css";

            graph.addNode(from);
            
            return processor.process(
                `.a :external(booga from "./local.css") { color: red; }`,
                { from, graph, resolve }
            )
            .then(() => expect(graph.overallOrder()).toMatchSnapshot());
        });

        it("should return useful errors when parsing", () => {
            var graph = new Graph(),
                from  = "packages/core/test/specimens/simple.css";

            graph.addNode(from);
            
            return processor.process(
                `@value sm, md, lg "../../shared.css";`,
                { from, graph, resolve }
            )
            .catch((e) => expect(e.message).toMatch(`SyntaxError: Expected "from"`));
        });
    });
});
