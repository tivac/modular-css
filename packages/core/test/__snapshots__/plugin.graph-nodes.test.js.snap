// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/plugins /graph-nodes.js should populate a graph with :external references 1`] = `
Array [
  "packages/core/test/specimens/local.css",
  "packages/core/test/specimens/simple.css",
]
`;

exports[`/plugins /graph-nodes.js should populate a graph with external @value references 1`] = `
Array [
  "packages/core/test/specimens/local.css",
  "packages/core/test/specimens/simple.css",
]
`;

exports[`/plugins /graph-nodes.js should populate a graph with external composes references 1`] = `
Array [
  "packages/core/test/specimens/local.css",
  "packages/core/test/specimens/simple.css",
]
`;
